### ISSUE TITLE
Corrupted docx file

### ISSUE THREAD

**Issue by symeonmattes** [plutext/docx4j#440]
Hi,

I have an issue with the docx4j library. The issue appears with docx4j 3.3.0, 3.36, 6.1.0 and java 8 and word from Microsoft 365 MSO (16.0.13801.20240, 32-bit

I have the following code:
```
InputStream in = ...     
ByteArrayOutputStream out = new ByteArrayOutputStream();             
WordprocessingMLPackage wordMLPackage;
wordMLPackage = WordprocessingMLPackage.load(in);
MainDocumentPart documentPart = wordMLPackage.getMainDocumentPart();

...

Docx4J.save(wordMLPackage, out);
```

This works just fine. However if I do this:
`for (Object p: documentPart .getContent()) {}`

Then when I open the file I receive the message:

> Word found unreadable content in ... Do you want to recover the contents of this document? If you trust the source of this document, click Yes.

I checked the document.xml of the corrupted file and the xml is validated properly. However when I open the corrupted file, let word fix it, save it and compare the fixed document.xml with the corrupted, only the schemas have changed. It seems that docx4j changes things from word.

```
corrupted:
xmlns:w="http://schemas.openxmlformats.org/wordprocessingml/2006/main" 
xmlns:m="http://schemas.openxmlformats.org/officeDocument/2006/math" 
xmlns:w14="http://schemas.microsoft.com/office/word/2010/wordml" 
xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships" 
xmlns:wp="http://schemas.openxmlformats.org/drawingml/2006/wordprocessingDrawing" 
xmlns:wp14="http://schemas.microsoft.com/office/word/2010/wordprocessingDrawing" 
xmlns:w15="http://schemas.microsoft.com/office/word/2012/wordml" 
xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
xmlns:wne="http://schemas.microsoft.com/office/word/2006/wordml" 
xmlns:o="urn:schemas-microsoft-com:office:office" 
xmlns:v="urn:schemas-microsoft-com:vml" 
xmlns:w10="urn:schemas-microsoft-com:office:word" 
xmlns:wps="http://schemas.microsoft.com/office/word/2010/wordprocessingShape" 
xmlns:w16se="http://schemas.microsoft.com/office/word/2015/wordml/symex" 
xmlns:w16cid="http://schemas.microsoft.com/office/word/2016/wordml/cid" 
mc:Ignorable="w14 w15 w16se w16cid w16 w16cex wp14"

Extra that currupted version has but not fixed from word
xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main" 
xmlns:sl="http://schemas.openxmlformats.org/schemaLibrary/2006/main" 
xmlns:c="http://schemas.openxmlformats.org/drawingml/2006/chart" 
xmlns:cdr="http://schemas.openxmlformats.org/drawingml/2006/chartDrawing" 
xmlns:c14="http://schemas.microsoft.com/office/drawing/2007/8/2/chart" 
xmlns:dgm="http://schemas.openxmlformats.org/drawingml/2006/diagram" 
xmlns:pic="http://schemas.openxmlformats.org/drawingml/2006/picture" 
xmlns:xdr="http://schemas.openxmlformats.org/drawingml/2006/spreadsheetDrawing" 
xmlns:dsp="http://schemas.microsoft.com/office/drawing/2008/diagram" 
xmlns:xvml="urn:schemasmicrosoftcom:office:excel" 
xmlns:pvml="urn:schemasmicrosoftcom:office:powerpoint" 
xmlns:cppr="http://schemas.microsoft.com/office/2006/coverPageProps" 
xmlns:odx="http://opendope.org/xpaths" 
xmlns:odc="http://opendope.org/conditions" 
xmlns:odq="http://opendope.org/questions" 
xmlns:oda="http://opendope.org/answers" 
xmlns:odi="http://opendope.org/components" 
xmlns:odgm="http://opendope.org/SmartArt/DataHierarchy" 
xmlns:b="http://schemas.openxmlformats.org/officeDocument/2006/bibliography" 
xmlns:wetp="http://schemas.microsoft.com/office/webextensions/taskpanes/2010/11" 
xmlns:we="http://schemas.microsoft.com/office/webextensions/webextension/2010/11" 
xmlns:comp="http://schemas.openxmlformats.org/drawingml/2006/compatibility" 
xmlns:lc="http://schemas.openxmlformats.org/drawingml/2006/lockedCanvas" 

Not corrupted, fixed by word:
xmlns:w="http://schemas.openxmlformats.org/wordprocessingml/2006/main" 
xmlns:m="http://schemas.openxmlformats.org/officeDocument/2006/math" 
xmlns:w14="http://schemas.microsoft.com/office/word/2010/wordml"
xmlns:r="http://schemas.openxmlformats.org/officeDocument/2006/relationships" 
xmlns:wp="http://schemas.openxmlformats.org/drawingml/2006/wordprocessingDrawing" 
xmlns:wp14="http://schemas.microsoft.com/office/word/2010/wordprocessingDrawing" 
xmlns:w15="http://schemas.microsoft.com/office/word/2012/wordml" 
xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
xmlns:wne="http://schemas.microsoft.com/office/word/2006/wordml" 
xmlns:o="urn:schemas-microsoft-com:office:office" 
xmlns:v="urn:schemas-microsoft-com:vml" 
xmlns:w10="urn:schemas-microsoft-com:office:word" 
xmlns:wps="http://schemas.microsoft.com/office/word/2010/wordprocessingShape" 
xmlns:w16se="http://schemas.microsoft.com/office/word/2015/wordml/symex" 
xmlns:w16cid="http://schemas.microsoft.com/office/word/2016/wordml/cid" 
mc:Ignorable="w14 w15 w16se w16cid w16 w16cex wp14"
---Extra from word that do not exist in corrupted---------
xmlns:wpc="http://schemas.microsoft.com/office/word/2010/wordprocessingCanvas" 
xmlns:cx="http://schemas.microsoft.com/office/drawing/2014/chartex" 
xmlns:cx1="http://schemas.microsoft.com/office/drawing/2015/9/8/chartex" 
xmlns:cx2="http://schemas.microsoft.com/office/drawing/2015/10/21/chartex" 
xmlns:cx3="http://schemas.microsoft.com/office/drawing/2016/5/9/chartex" 
xmlns:cx4="http://schemas.microsoft.com/office/drawing/2016/5/10/chartex" 
xmlns:cx5="http://schemas.microsoft.com/office/drawing/2016/5/11/chartex" 
xmlns:cx6="http://schemas.microsoft.com/office/drawing/2016/5/12/chartex" 
xmlns:cx7="http://schemas.microsoft.com/office/drawing/2016/5/13/chartex" 
xmlns:cx8="http://schemas.microsoft.com/office/drawing/2016/5/14/chartex" 
xmlns:aink="http://schemas.microsoft.com/office/drawing/2016/ink" 
xmlns:am3d="http://schemas.microsoft.com/office/drawing/2017/model3d" 
xmlns:w16cex="http://schemas.microsoft.com/office/word/2018/wordml/cex" 
xmlns:w16="http://schemas.microsoft.com/office/word/2018/wordml" 
xmlns:wpg="http://schemas.microsoft.com/office/word/2010/wordprocessingGroup" 
xmlns:wpi="http://schemas.microsoft.com/office/word/2010/wordprocessingInk" 

```

Any ideas how to fix this?

**Comment by plutext** [comment]
See also https://github.com/plutext/docx4j/issues/381  Fixed in 8.1.5.  Any reason you can't upgrade?

**Comment by symeonmattes** [comment]
No...reason. I haven't tried it. In maven I can see that it goes till 6.1.2 ([docx4j](https://mvnrepository.com/artifact/org.docx4j/docx4j))

**Comment by plutext** [comment]
Please see https://www.docx4java.org/downloads.html which says

docx4j is in Maven Central: Select one and only one of:

- docx4j-JAXB-Internal (available for docx4j 8.1.x only),
- docx4j-JAXB-MOXy (for docx4j 8 or 11),
- docx4j-JAXB-ReferenceImpl (for docx4j 8 or 11),

**Comment by symeonmattes** [comment]
Thanks it worked...I used docx4j-JAXB-MOXy and it works fine with 8.1.7
